### PR TITLE
switch to StaticArrays and get 0.6 working [WIP]

### PR DESCRIPTION
**Pull request by SimonDanisch**

### CODE REVIEW

**Comment by SimonDanisch** [comment]
This disables convexhulls and gjk for now, since they error and I don't really use that part of GeometryTypes.
I hope @jw3126 has time to look into those :)

**Comment by SimonDanisch** [comment]
Also, seems like relying on StaticArrays makes GeometryTypes 0.6 only -.- ... not sure if I like that, but maybe reasonable, considering that people can tag a previous version for 0.5!

**Comment by jw3126** [comment]
I will fix the gjk stuff, when I switch to v0.6.

**Comment by jw3126** [comment]
Should I fork this branch or better wait until its merged with master?

**Comment by SimonDanisch** [comment]
hm good question! I'm a bit hesitant with merging it, since it makes geometrytypes 0.6 only... But maybe it's time for that! Note that you'll need: https://github.com/JuliaArrays/StaticArrays.jl/pull/138
I should just try to merge at least that into master as quick as possible!

**Comment by jw3126** [comment]
No rush from my side.

**Comment by sbromberger** [comment]
> maybe it's time for that

Yes. It's time! :)

(I really miss being able to plot stuff in 0.6.)

File: src/lines.jl [hunk 1]
```diff
@@ -45,23 +48,92 @@ end
...
+    result = Vector{P}(length(range)+1)
     for (i,j) in enumerate(range)
         result[i] = vec[mod1(j, length(vec))]
     end
     result[end] = endpoint
     result
 end
 
+
+# Taken from Iterators.jl, since it's not possible to use Iterators.jl on 0.6 with precompilation
```
Comment (tkelman) [inline src/lines.jl]: when you copy code, you need to respect the license of that code

Comment (SimonDanisch) [inline src/lines.jl]: oh, that was just supposed to be a local stop gap solution. Is there actually a fix by now? I just tried it right now and am still getting a precompile error on first precompile when importing Iterators....

Comment (tkelman) [inline src/lines.jl]: try on rc3

Comment (SimonDanisch) [inline src/lines.jl]: ha can't reproduce it anymore :) I removed the code in https://github.com/JuliaGeometry/GeometryTypes.jl/pull/100

**Pull request merged**